src/wallet/wallet2.cpp

```diff
@@ -11448,29 +11529,48 @@ uint64_t wallet2::import_key_images(const std::vector<std::pair<crypto::key_imag
   return m_transfers[signed_key_images.size() - 1].m_block_height;
 }
 
-bool wallet2::import_key_images(std::vector<crypto::key_image> key_images)
+bool wallet2::import_key_images(std::vector<crypto::key_image> key_images, size_t offset, boost::optional<std::unordered_set<size_t>> selected_transfers)
 {
-  if (key_images.size() > m_transfers.size())
+  if (key_images.size() + offset > m_transfers.size())
```

Comment:
Is this offset coming from a trusted source (ie, the code) or the device ? It the latter, this should check for overflow.

Reply:
The offset is from the wallet. It is like the offset that is generated by `wallet2::export_outputs`